File: backend/src/main/java/com/redscooter/API/order/DTO/CreateOrderDTO.java
package com.redscooter.API.order.DTO;

import com.redscooter.API.OrderLine.DTO.CreateOrderLineDTO;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CreateOrderDTO {
    @NotEmpty
    @Valid
    private List<CreateOrderLineDTO> orderLines;

    @Valid
    private CreateOrderBillingDTO orderBilling;

    private Long userId; // optional
}

File: frontend/src/services/models/CreateOrderDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { CreateOrderBillingDTO } from './CreateOrderBillingDTO';
import type { CreateOrderLineDTO } from './CreateOrderLineDTO';

export type CreateOrderDTO = {
    orderLines: Array<CreateOrderLineDTO>;
    orderBilling?: CreateOrderBillingDTO;
    userId?: number;
};
